src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -1124,7 +1124,8 @@ private void OnAsyncTaskStateChanged(DynamoScheduler sender, TaskStateChangedEve
                         long end = e.Task.ExecutionEndTime.TickCount;
                         var executionTimeSpan = new TimeSpan(end - start);
 
-                        if (Logging.Analytics.ReportingAnalytics)
+                        //don't attempt to send these events unless GA is active or ADP will actually record these events.
+                        if (Logging.Analytics.ReportingGoogleAnalytics || AnalyticsService.IsADPCollectionEnabled(Actions.Run,Categories.NodeOperations))
```

Comment:
is there a significant performance in checking `IsADPCollectionEnabled` here...rather than checking it directly in Analytics.NET.ADP ?
Deferring the check to Analytics.NET.ADP would simplify the changes in Dynamo...but might be a bit slower...

Reply:
Sorry @pinzart90 I'm not sure what you mean - is what you mean here similar to your suggestion below to check `IsADPCollectionEnabled` in the `TrackEvent` calls?

If so - this is not what the sdk docs suggest. I've looked at the source a bit and I'm not sure what the performance impact could be - maybe we can look it over together.
My worry is that since the api docs say it checks if the user is opted out, and that the call is blocking - that in the worst case this might mean we inadvertently introduce a blocking call that waits on a server into every single track event call.

Reply:
` is what you mean here similar to your suggestion below to check IsADPCollectionEnabled in the TrackEvent calls?`
Yes